com.ibm.wala.cast.python.jython3/source/com/ibm/wala/cast/python/parser/PythonModuleParser.java

```diff
@@ -70,17 +160,47 @@ public CAstNode visitImportFrom(ImportFrom importFrom) throws Exception {
 
         if (s.isPresent()) {
           String moduleName = s.get();
-          LOGGER.finer("Module name from " + importFrom + " is: " + moduleName + ".");
+          LOGGER.finer("Module name from " + importFrom + " is: " + moduleName);
+
+          if (moduleName.startsWith(".")) {
+            LOGGER.info("Found relative import: " + moduleName);
+            moduleName = this.resolveRelativeImport(moduleName);
+          }
 
-          if (!isLocalModule(moduleName)) {
-            LOGGER.finer("Module: " + moduleName + " isn't local.");
-            moduleName = s.get() + "/__init__";
-          } else LOGGER.finer("Module: " + moduleName + " is local.");
+          if (!isLocalModule(moduleName)) moduleName += "/" + MODULE_INITIALIZATION_ENTITY_NAME;
 
-          LOGGER.finer("Module name from " + importFrom + " is: " + moduleName + ".");
+          LOGGER.finer("Module name from " + importFrom + " is: " + moduleName);
 
           if (isLocalModule(moduleName)) {
-            LOGGER.finer("Module: " + moduleName + ".py" + " is local.");
+            List<File> pythonPath = PythonModuleParser.this.getPythonPath();
+            LOGGER.info("PYTHONPATH is: " + pythonPath);
+
+            // If there is a PYTHONPATH specified.
+            if (pythonPath != null && !pythonPath.isEmpty()) {
+              // Adjust the module name per the PYTHONPATH.
+              Optional<SourceModule> localModule = getLocalModule(moduleName);
+
+              for (File pathEntry : pythonPath) {
+                Path modulePath =
+                    localModule
+                        .map(SourceModule::getURL)
+                        .map(URL::getFile)
+                        .map(Path::of)
+                        .orElseThrow(IllegalStateException::new);
+                LOGGER.finer("Found module path: " + modulePath);
+
+                if (modulePath.startsWith(pathEntry.toPath())) {
+                  // Found it.
+                  Path scriptRelativePath = pathEntry.toPath().relativize(modulePath);
+                  LOGGER.finer("Relativized path is: " + scriptRelativePath);
+
+                  // Remove the file extension if it exists.
+                  moduleName = scriptRelativePath.toString().replaceFirst("\\.py$", "");
+                  LOGGER.fine("Using module name: " + moduleName);
+                  break;
+                }
```

Comment:
This code seems a bit duplicated with code earlier in the PR?  Is it worth extracting a method?

Reply:
Good catch, thanks. I'll have a look.

Reply:
It's a little difficult to fully extract this since it's in a loop with a `break`. I think i can extract some of it at least.

Reply:
I gave it a shot.